Kernel/Devices/GPU/3dfx/VoodooDisplayConnector.cpp

```diff
@@ -0,0 +1,411 @@
+        return result;
+    }
+
+    dbgln("3dfx: Looking for suitable mode with resolution {}x{}", width, height);
+
+    IterationDecision iteration_decision = TRY(m_edid_parser->for_each_detailed_timing([&](auto& timing, unsigned) {
+        dbgln("3dfx: Considering detailed timing {}x{} @ {}", timing.horizontal_addressable_pixels(), timing.vertical_addressable_lines(), timing.refresh_rate());
+
+        if (timing.is_interlaced() || timing.horizontal_addressable_pixels() != width || timing.vertical_addressable_lines() != height) {
+            return IterationDecision::Continue;
+        }
+
+        result.pixel_clock_in_khz = timing.pixel_clock_khz();
+        result.horizontal_front_porch_pixels = timing.horizontal_front_porch_pixels();
+        result.horizontal_sync_time_pixels = timing.horizontal_sync_pulse_width_pixels();
+        result.horizontal_blank_pixels = timing.horizontal_blanking_pixels();
+        result.vertical_front_porch_lines = timing.vertical_front_porch_lines();
+        result.vertical_sync_time_lines = timing.vertical_sync_pulse_width_lines();
+        result.vertical_blank_lines = timing.vertical_blanking_lines();
+        return IterationDecision::Break;
+    }));
+
+    if (iteration_decision == IterationDecision::Break) {
+        return result;
+    }
+
+    iteration_decision = TRY(for_each_dmt_timing_in_edid([&](auto& timing) {
+        dbgln("3dfx: Considering DMT timing {}x{} @ {}", timing.horizontal_pixels, timing.vertical_lines, timing.vertical_frequency_hz());
+
+        if (timing.scan_type != EDID::DMT::MonitorTiming::ScanType::NonInterlaced || timing.horizontal_pixels != width || timing.vertical_lines != height) {
+            return IterationDecision::Continue;
+        }
+
+        result.pixel_clock_in_khz = timing.pixel_clock_khz;
+        result.horizontal_front_porch_pixels = timing.horizontal_front_porch_pixels;
+        result.horizontal_sync_time_pixels = timing.horizontal_sync_time_pixels;
+        result.horizontal_blank_pixels = timing.horizontal_blank_pixels;
+        result.vertical_front_porch_lines = timing.vertical_front_porch_lines;
+        result.vertical_sync_time_lines = timing.vertical_sync_time_lines;
+        result.vertical_blank_lines = timing.vertical_blank_lines;
+        return IterationDecision::Break;
+    }));
+
+    if (iteration_decision == IterationDecision::Break) {
+        return result;
+    }
+
+    dbgln("3dfx: No timing information available for display mode {}x{}", width, height);
+    return EINVAL;
+}
+
+ErrorOr<void>
+VoodooDisplayConnector::set_mode_setting(ModeSetting const& requested_mode_setting)
+{
+    SpinlockLocker locker(m_modeset_lock);
+    VERIFY(m_framebuffer_console);
+
+    ModeSetting mode_setting = TRY(find_suitable_mode(requested_mode_setting));
+    dbgln("VoodooDisplayConnector resolution registers set to - {}x{}", mode_setting.horizontal_active, mode_setting.vertical_active);
```

Comment:
Please put this behind a debug flag (something like `dbgln_if(3DFX_DEBUG, ...)` is good here).

Reply:
I've put the dbglns behind the debug flag (and rebased+retested). I've used `TDFX_DEBUG` because it can't start with a 3.